### ISSUE TITLE
'.button-link' color contrast issue, lack of hover state, should probably be a link in some cases

### ISSUE THREAD

**Issue by LZoog** [mozilla/fxa#3951]
I noticed while working on #3843 that:
- The color we set for `.button-link` does not pass ADA color contrast requirements. It's set to `$color-blue` or `#0a84ff`, which fails for "normal" text (bolded text [below 14pt/18.66px](https://webaim.org/resources/contrastchecker/)). Our normal link color is `#0060df` which does pass. They should probably be this color.
- We don't have a hover state for these. They should underline and darken with a `150ms` transition like our normal links do.
- Thirdly, let's double check all instances of `.button-link` should actually be links and not buttons. At the very least, they should be styled differently. When you hover over one of these, since the button takes up the entire width, you can click in whitespace left on the left/right-hand side and interact with it.

Here's one example, this should probably be a link as it just redirects to `/signin`.
<img src="https://user-images.githubusercontent.com/13018240/72850997-0ff5eb80-3c70-11ea-8399-f351241fb1f7.png" width=400>

previous page with just a link:
<img width="400" alt="image" src="https://user-images.githubusercontent.com/13018240/72851456-4a13bd00-3c71-11ea-94e6-678bc00600b0.png">

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/FXA-967)
┆Issue Number: FXA-967

**Comment by vasilicatamas** [comment]
I confirm that all "Remember password? Sign in", "Don't have a recovery key" and "Continue to my account" buttons appears as links, have hover state and #0060df color.  

@LZoog I think "Cancel" button displayed in first modal from generate recovery key flow should also follow these rules - filed #6166 

![Screenshot_6](https://user-images.githubusercontent.com/18114210/89668694-e12b2400-d8e6-11ea-8135-44ed1b996983.png)
![Screenshot_7](https://user-images.githubusercontent.com/18114210/89668698-e1c3ba80-d8e6-11ea-8007-3b7ce91ae9b7.png)
![Screenshot_8](https://user-images.githubusercontent.com/18114210/89668699-e25c5100-d8e6-11ea-97ea-f2b8eb935254.png)
![Screenshot_9](https://user-images.githubusercontent.com/18114210/89668702-e25c5100-d8e6-11ea-978f-c74722cf44a0.png)

Based on the above mentioned I am marking this bug as verified.
